File: 44_fizzbuzz/cpp/main.cpp
#include <iostream>
#include <charconv>
//#include <cstdio>

using namespace std;

auto main(int argc, char **argv) -> int {

  if (argc != 2) {
    std::cerr << "error: one numeric input is required\n";
    return 1;
  }

  int limit{};
  auto [ptr, ec] = std::from_chars(argv[1], argv[1] + sizeof(argv[1]), limit);

  if (ec == std::errc()) {

    for(int n = 0; n < limit; ++n) {
      if (n % 15 == 0) {
        std::cout << "fizzbuzz\n";
      } else if (n % 5 == 0) {
        std::cout << "fizz\n";
      } else if (n % 3 == 0) {
        std::cout << "buzz\n";
      } else {
        std::cout << n << "\n";
      }
    }

  } else {
    std::cout << "error: input not a valid number\n";
    return 1;
  }

  return 0;
}

File: 44_fizzbuzz/rust/src/main.rs
fn main() {
    match std::env::args().len() {
        // one argument passed
        2 => {
            let input: u16 = match std::env::args()
                .nth(1)
                .expect("error: could not get argument")
                .parse()
            {
                Ok(n) => n,
                Err(_) => {
                    eprintln!("error: need a positive numeric input");
                    std::process::exit(2)
                }
            };

            for n in 0..input {
                if n % 15 == 0 {
                    println!("fizzbuzz");
                } else if n % 5 == 0 {
                    println!("fizz");
                } else if n % 3 == 0 {
                    println!("buzz");
                } else {
                    println!("{}", n);
                }
            }
        }
        // none, and all other arguments passed
        _ => {
            eprintln!("error: only one input allowed");
            std::process::exit(2)
        }
    }
}
